src/main/java/ca/bc/gov/ag/efax/mail/scheduled/EmailPoller.java

```diff
@@ -40,13 +44,21 @@ public void pollForEmails() throws Exception {
         for (EmailMessage emailMessage : emailService.getInboxEmails()) {
             
             // Attempt to parse the email response from MS Exchange
-            DocumentDistributionMainProcessProcessResponse response = emailParser.parse(emailMessage);
+            DocumentDistributionMainProcessProcessResponseDecorator response = emailParser.parse(emailMessage);
 
             // If the jobId is blank, there's nothing we can do except move on. (We can't send a message to the Justin Callback informing the user
             // that a message with a certain jobId succeeded or failed).  If so, an error message should have been logged for review so the email
             // parser can be improved upon.
             if (!StringUtils.isEmpty(response.getJobId())) {                     
                 documentDistributionService.sendResponseToCallback(response);
+                
+                // remove the referenced message from the redis queue
+                try {
+                    sentMessageRepository.deleteById(response.getUuid());
```

Comment:
47, 58-61 might not be tested

Reply:
Yes, good observation.  I'm just writing some units tests for this draft PR change.